### PR TITLE
Data Science with Python : Logistic Regression

### LINKED ISSUE #1125: Data Science with Python : Logistic Regression Algorithm
Welcome to 'DSWP' Team, good to see you here.

This issue will helps readers in acquiring all the knowledge to learn about getting involved with Logistic Regression Algorithm. Tutorial to Logistic Regression Algorithm and how it's applied using sample code.

To get assigned to this issue, add your serial numbers mentioned in the spreadsheet of "Data Science with Python", the approach one would follow and choice you prefer (Documentation, Audio, Video). You can go with all three or any number of options you're interested to work on.

This issue is created for beginners and who aren't assigned to any issues yet.

If you had referred any resources, add them up in "DS Resources". Similarly if you had used datasets, include them in "DS Datasets".

Domain : Machine Learning

Mentors Assigned : [Aishani Singh](https://github.com/aishanii) / [Yash Goel](https://github.com/Yash621)

Points to Note :

The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.

"Issue Title" and "PR Title should be the same. Include issue number along with it.

Changes should be made inside the Datascience_With_Python/ directory & Datascience_With_Python branch.

Follow Contributing Guidelines & Code of Conduct before start Contributing.

This issue is only for 'GWOC' contributors of 'DSWP' domain.

All the best. Enjoy your open source journey ahead. 😎

### PR DESCRIPTION
**Pull request by rahulsinghal1904**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #1125

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [x] Datascience with Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [X ] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [X ] I have performed a self-review of my own code or work.
- [X ] I have commented my code, particularly in hard-to-understand areas.
- [ X] My changes generates no new warnings.
- [X ] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md [hunk 1]
```diff
@@ -40,15 +40,26 @@ In Logistic regression, sigmoid function which is an 'S' shaped graph is used fo
...
 
 hΘ(x) = sigmoid (Z)
+
 where sigmoid function is given by:
 
 sigmoid(t)=1/(1+e^-t)
+
+<p align="center">
+  <img src="Sigmoid Curve.png" width="350" title="sigmoid curve">
+</p>
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md:54]: I'm not able to view the images that have been added in the README file, @prathimacode-hub are you able to view the images that have been added?

File: Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md [hunk 2]
```diff
@@ -40,15 +40,26 @@ In Logistic regression, sigmoid function which is an 'S' shaped graph is used fo
...
 We then specify a decision boundary according to which the values are classified into particular classes. After this, using the predicted values we can give different classes to the target variable.
 
 ## Workflow of the algorithm
 
+<p align="center">
+  <img src="Code flowchart.jpeg" width="350" title="flowchart of code example">
+</p>
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md:61]: Try dropping the images directly in the README file where they are required

**Review by aishanii — COMMENTED** [review]
Images not visible in the README, please check!

File: Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md (hunk 2 above)
Comment (rahulsinghal1904) [reply Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md]: I have updated the README file, please confirm if the images are visible to you now?
Thank you!

**Comment by rahulsinghal1904** [comment]
Please let me know if any other changes are required.

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
Changes to be done :

Sorry to say, but this is not sufficient, Especially being the algorithm of ML, Practical implementation using sample test case is needed too justify this algorithm
And add all the images used in "Images" folder in this repo.

@rahulsinghal1904

**Comment by rahulsinghal1904** [comment]
@prathimacode-hub Thanks for the update. I have added the images in the Images folder. Also, I have already provided the implementation of code in the Code Implementation.ipynb file. Can you please look into it and suggest if any further changes are required? Thanks!

File: Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md [hunk 3]
```diff
@@ -40,15 +40,23 @@ In Logistic regression, sigmoid function which is an 'S' shaped graph is used fo
...
 This Z is passed into the sigmoid function
 
 hΘ(x) = sigmoid (Z)
+
 where sigmoid function is given by:
 
 sigmoid(t)=1/(1+e^-t)
+
+![Sigmoid Curve](https://user-images.githubusercontent.com/54388275/135406879-737a0377-180d-4d82-ac9d-970075ec8c66.png)
+
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Algorithms/Logistic Regression/README.md:53]: Yes they are visible now

**Review by aishanii — COMMENTED** [review]
@prathimacode-hub I've reviewed the PR, please confirm and merge

**Review by prathimacode-hub — APPROVED** [review]
PR approved. Good work. 👍👌@rahulsinghal1904

**Pull request merged**